### PR TITLE
Orbital Ships

### PR DESCRIPTION
**Pull request by Leibovitch**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,8 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+tester.py
+tile_dictionary.json
+fetch_parameters.json
+decrypted_files
```
Comment (mzedayda) [inline .gitignore]: decrypton?

File: aws_objects_integration.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                                my_object['area'] = area    
+                                my_object['width']= width
+                                my_object['length'] = length
+                                my_object['tile_id'] = '' 
+                                ships_list.append(my_object) 
+
+                        elif detection['geometry']['type'] == 'Polygon':
+                            area = Polygon(detection['geometry']['coordinates']).area
+                            if len(detection['geometry']['coordinates']) == 4:
+                                dim1 = ml(detection['geometry']['coordinates'][0] * ureg.degree, detection['geometry']['coordinates'][1] * ureg.degree)
```
Comment (mzedayda) [inline aws_objects_integration.py]: don't use unclear shorthand, ml

Comment (mzedayda) [reply aws_objects_integration.py]: Use distances p0-p1 and p1-p2..  safer..

File: aws_objects_integration.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                                dim2 = ml(detection['geometry']['coordinates'][0] * ureg.degree, detection['geometry']['coordinates'][3] * ureg.degree)
+                                if (0 in detection.geoemetry['coordinates'][1] - detection.geoemetry['coordinates'][0]):
```
Comment (mzedayda) [inline aws_objects_integration.py]: in?

Comment (mzedayda) [reply aws_objects_integration.py]: What is `detection.geoemetry['coordinates'][1]`?

File: aws_objects_integration.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                                    length = math.sqrt(dim1 ** 2 + dim2 ** 2)
+                                    width = area / length
+                                else: 
+                                    length = max(dim1, dim2)
+                                    width = max(dim1, dim2)
```
Comment (mzedayda) [inline aws_objects_integration.py]: both max?

File: aws_objects_integration.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+                            my_object['originalImageId'] = scene_id
+                            my_object['observed_start'] = imaging_time
+                            my_object['observed_end'] = imaging_time
+                            my_object['area'] = area    
+                            my_object['width']= width
+                            my_object['length'] = length
+                            my_object['tile_id'] = '' 
+                            ships_list.append(my_object)
+                                
+                        elif detection['geometry']['type'] == 'MuliPoint':
```
Comment (mzedayda) [inline aws_objects_integration.py]: Muli?

File: aws_objects_integration.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
+import os
+import sys
+import json
+import pymongo
+import boto3
+import pprint
+import math
+import numpy as np
+import sys
+from pint import UnitRegistry as ur
```
Comment (mzedayda) [inline aws_objects_integration.py]: No need fot ur

File: aws_objects_integration.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+from shapely.geometry import Polygon
+from geocalc import metric_length as ml
```
Comment (mzedayda) [inline aws_objects_integration.py]: No need for ml

File: aws_objects_integration.py [hunk 8]
```diff
@@ -0,0 +1,164 @@
+import os
+import sys
+import json
+import pymongo
+import boto3
+import pprint
```
Comment (mzedayda) [inline aws_objects_integration.py]: Never used

File: aws_objects_integration.py [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+# getting parameters from user and parsing them
+# sys.argv[1] is the path to the config file that determines where and when to download from
+
+filename, file_extension = os.path.splitext(sys.argv[1])
+if file_extension == '.json':
+    input_configuration = json.load(open(sys.argv[1]))
+elif file_extension == '.txt':
+    input_configuration = decrypt_to_json(sys.argv[1])
+
+AOI_polygon = Polygon(input_configuration['stats_endpoint_request']['geometry']['coordinates'][0])
```
Comment (mzedayda) [inline aws_objects_integration.py]: That's a point?

File: encryptions/decrypt_to_json.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    file_name = list_file_path[len(list_file_path)- 1]
+    file_name_no_ext = os.path.splitext(file_name)[0]
+    original_name =  file_name_no_ext.split('_')
+    original_name.pop()
+    original_name = '_'.join(original_name)
+    key = os.environ[original_name.upper() + '_KEY']
+    enc_dec = Fernet(key)
+    encrypt_text = open(file_path).read()
+    decrypted_text = enc_dec.decrypt(encrypt_text.encode()).decode()
+    return json.load(decrypted_text)
```
Comment (mzedayda) [inline encryptions/decrypt_to_json.py]: What and why are you encrypting?

File: geocalc.py [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    y2 = math.sin(point2[0]) * math.cos(point2[1])
+    z2 = math.sin(point2[1])
+
+    pos1 = [x1 ,y1 ,z1]
+    pos2 = [x2 ,y2, z2]
+
+    distance = config['earth_radius'] * math.acos(np.dot(pos1, pos2)) * ureg.meter
+    return distance.to('kilometer')
+
+def get_coordintes(row, col):
```
Comment (mzedayda) [inline geocalc.py]: typo

File: geocalc.py [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    y1 = math.sin(point1[0]) * math.cos(point1[1])
+    z1 = math.sin(point1[1])
+    x2 = math.cos(point2[0]) * math.cos(point2[1])
+    y2 = math.sin(point2[0]) * math.cos(point2[1])
+    z2 = math.sin(point2[1])
+
+    pos1 = [x1 ,y1 ,z1]
+    pos2 = [x2 ,y2, z2]
+
+    distance = config['earth_radius'] * math.acos(np.dot(pos1, pos2)) * ureg.meter
```
Comment (mzedayda) [inline geocalc.py]: earth radius is not configurable. make it hard coded constant.

Ideally you should create constants.py (EARTH_RADIUS, E, PI) in _pyCoral_ and use it here

File: geocalc.py [hunk 13]
```diff
@@ -0,0 +1,36 @@
+import math
+import json
+import pint
+import numpy as np
+
+
+config = json.load(open('config.json'))
+ureg = pint.UnitRegistry()
+# print(config)
```
Comment (mzedayda) [inline geocalc.py]: Don't leave comments in PRs

File: geocalc.py [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+reference_row = config['reference_row']
+reference_col = config['reference_col']
+step = config['step']
+
+def metric_length(point1, point2):
+    point1 = np.deg2rad(point1)
+    point2 = np.deg2rad(point2)
+    #input should be a pint quantity of either degrees or angles
```
Comment (mzedayda) [inline geocalc.py]: So enforce it.

This method should move to pyCoral, but for similar things in the future

File: get_tile_dictionary.py [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+config = json.load(open('config.json'))
+bucket_name = config['aws_ships_bucket_name']
+ACCESS_KEY = os.environ['ORBITAL_AWS_ACCESS_KEY']
+SECRET_KEY = os.environ['ORBITAL_AWS_SECRET_KEY']
+
+s3 = boto3.resource('s3', aws_access_key_id=ACCESS_KEY, aws_secret_access_key=SECRET_KEY)
+bucket = s3.Bucket(bucket_name)
+index = bucket.Object(key='tile_index.geojson')
+
+iterable_index = json.load(index.get()['Body'])
```
Comment (mzedayda) [inline get_tile_dictionary.py]: Why the complicated name? this is just the tile index. so call it tile_index

File: get_tile_dictionary.py [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+#         print(prev_col)
+#     last_col = col_dict[col]
+#     prev_col = col
+
+# pp.pprint(col_dict)
+pp.pprint(tile_dict)
+
+# import json
+# with open('tile_dictionary.json', 'w') as fp:
+#     json.dump(tile_dict, fp)
```
Comment (mzedayda) [inline get_tile_dictionary.py]: Who uses this file, everything is commented out.
PRs cannot be submitted like that

File: aws_objects_integration.py [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+db = client.louvre
+ship_collection = db.ship
+ships_list = [] 
+bad_files = []
+bad_files_log = open('bad_files_log','w+')
+bad_files_log.write('the following files cannot be used bacause of the specified reason \n\n')
+
+for day in range(delta + 1): 
+    for AOI in orbital_AOIS:
+        prefix = 'ship_detection/' + AOI + '/' + (start_time + timedelta(day)).strftime('%Y-%m-%d')
```
Comment (mzedayda) [inline aws_objects_integration.py]: `prefix = 'ship_detection/{}/{}'.format(aoi, day.strftime('%Y-%m-%d'))`

File: aws_objects_integration.py [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+client = pymongo.MongoClient('mongodb://localhost:27017/') #just for testing purposes
+# client = pymongo.MongoClient(mongo_uri)
+db = client.louvre
+ship_collection = db.ship
+ships_list = [] 
+bad_files = []
+bad_files_log = open('bad_files_log','w+')
+bad_files_log.write('the following files cannot be used bacause of the specified reason \n\n')
+
+for day in range(delta + 1):
```
Comment (mzedayda) [inline aws_objects_integration.py]: day should be a datetime object:
`while day < end_date:`

File: aws_objects_integration.py (hunk 3 above)
Comment (Leibovitch) [reply aws_objects_integration.py]: i'm looking to see if the two first coordinates in rectangular polygon are horizontally aligned

**Pull request closed without merging**